crates/ruff_cli/src/diagnostics.rs

```diff
@@ -186,6 +202,7 @@ pub(crate) fn lint_path(
     noqa: flags::Noqa,
     fix_mode: flags::FixMode,
     unsafe_fixes: UnsafeFixes,
+    extension_override: &FxHashMap<String, Language>,
```

Comment:
nit: we could probably use a struct instead with a hidden field to hide the implementation details and provide a method to resolve the extension to the language. That way we could, if we want, change it or extend it in the future. This is totally fine as well :)

Reply:
done